spring-boot/admin/src/main/java/eu/coatrack/admin/service/OauthUserAccountManagement.java

```diff
@@ -0,0 +1,127 @@
+
+    private OAuth2User getAttributesFromAuthentication() {
+        OAuth2User loggedInUser = (OAuth2User) SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getPrincipal();
+        return loggedInUser;
+    }
+
+    public String getLoginNameFromLoggedInUser() {
+        return getAttributesFromAuthentication().getAttribute("login");
+    }
+
+    public String getNameFromLoggedInUser() {
+        return getAttributesFromAuthentication().getAttribute("name");
+    }
+
+    public String getCompanyFromLoggedInUser() {
+        return getAttributesFromAuthentication().getAttribute("company");
+    }
+
+    public String getEmailFromLoggedInUser() throws JsonProcessingException {
+        String email = getAttributesFromAuthentication().getAttribute("email");
+
+        // If the email is defined as private on GitHub (The Oauth2 will retrieve the
+        // attribute email as null),
+        // so the only way to retrieve is to make request with the Oauth Token
+        if (email == null || email.isEmpty()) {
+
+            ResponseEntity<String> emailListFromGithub = getEmailsListFromGithub();
+            email = GetPrimaryEmailFromLoggedInUser(emailListFromGithub);
+        }
+        return email;
+    }
+
+    private ResponseEntity<String> getEmailsListFromGithub () {
+        // Initialize RestTemplate and HTTP Headers
+        RestTemplate restTemplate = new RestTemplate();
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("Authorization", "token " + getTokenFromLoggedInUser());
+        HttpEntity<String> githubRequest = new HttpEntity(headers);
+
+        // Make the request to GitHub for the emails
+        ResponseEntity<String> userEmailsRequest = restTemplate.exchange(GITHUB_API_EMAIL, HttpMethod.GET,
+                githubRequest, String.class);
+        return userEmailsRequest;
+    }
+
+    private String GetPrimaryEmailFromLoggedInUser(ResponseEntity<String> userEmailsRequest) throws JsonProcessingException {
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+        List<GithubEmail> emailsList = objectMapper.
+                readValue(userEmailsRequest.getBody(),
+                        objectMapper
+                                .getTypeFactory()
+                                .constructCollectionType(List.class, GithubEmail.class));
+
+        for (GithubEmail userEmail : emailsList) {
+            if (userEmail.getVerified() == true && userEmail.getPrimary()) {
+                return userEmail.getEmail();
+            }
+        }
+        return null;
+    }
+
+    public String getTokenFromLoggedInUser() {
+        Authentication authentication = SecurityContextHolder
+                .getContext()
+                .getAuthentication();
+        OAuth2AuthenticationToken oauthToken = (OAuth2AuthenticationToken) authentication;
```

Comment:
Are we sure that this will always be an instance of "OAuth2AuthenticationToken"? Maybe we should add a try-catch block for a ClassCastException.

Reply:
This thread was discussed, and it was decided that a try catch block is not needed anymore.
So this is thread is resolved